esmvaltool/diag_scripts/monitor/multi_datasets.py

```diff
@@ -1862,6 +2209,140 @@ def _get_reference_dataset(self, datasets):
             return ref_datasets[0]
         return None
 
+    def _get_benchmarking_reference(self, datasets):
+        """Extract reference dataset for calculation of benchmarking metric."""
+        variable = datasets[0][self.cfg['group_variables_by']]
+        ref_datasets = [d for d in datasets if
+                        d.get('reference_for_metric', False)]
+
+        if len(ref_datasets) == 1:
+            return ref_datasets[0]
+
+        # try variable attribute "reference_dataset"
+        for dataset in datasets:
+            print(dataset.get('reference_dataset'))
+            print(dataset.get('dataset'))
+            if dataset.get('reference_dataset') == dataset.get('dataset'):
+                ref_datasets = dataset
+                break
+        if len(ref_datasets) != 1:
+            raise ValueError(
+                f"Expected exactly 1 reference dataset for variable "
+                f"'{variable}', got {len(ref_datasets)}")
+        return None
+
+    def _get_benchmark_datasets(self, datasets):
```

Comment:
Is there documentation (in code or docs) to indicate that user should specify `benchmark_dataset: true`  in recipe, to select that dataset to be benchmarked? This is clear from example recipes, but should really be captured somewhere beyond those.

Worth noting in that documentation that some plots will (I think) only plot a single benchmark even if multiple specified.

Reply:
Good point. I'll add that to the documentation.